src/pipe.ts

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable complexity, max-depth, max-lines, max-statements -- TODO: We should strive to simplify the pipe implementation! */
```

Comment:
**thought:** see, if we used iterators to deal with laziness, i think we can cut a lot of the complexity here

Reply:
but of course, it would require a somewhat-substantial rewriting of things